### PR TITLE
Added Cops&Crims and Arrow ModGUI Components

### PR DESCRIPTION
**Pull request by FalseHonesty**

### CODE REVIEW

File: src/main/java/eladkay/hudpixel/modulargui/components/ArrowCounterModularGuiProvider.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+import eladkay.hudpixel.modulargui.SimpleHudPixelModularGuiProvider;
+import net.minecraft.client.entity.EntityPlayerSP;
+import net.minecraft.init.Items;
+import net.minecraft.item.ItemStack;
+import net.minecraftforge.fml.client.FMLClientHandler;
+import net.unaussprechlich.hudpixelextended.util.McColorHelper;
+
+public class ArrowCounterModularGuiProvider extends SimpleHudPixelModularGuiProvider implements McColorHelper {
+
+    @ConfigPropertyBoolean(category = CCategory.HUD, id = "copsAndCrimsHeadshotCounter", comment = "Cops and Crims Headshot Counter", def = true)
```
Comment (unaussprechlich) [inline src/main/java/eladkay/hudpixel/modulargui/components/ArrowCounterModularGuiProvider.java:15]: This is not Cops and Crimes :P

File: src/main/java/eladkay/hudpixel/modulargui/components/ArrowCounterModularGuiProvider.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    }
+
+    @Override
+    public String getAfterstats() {
+        return null;
+    }
+
+    @Override
+    public boolean doesMatchForGame() {
+        return !GameDetector.getGameHasntBegan();
```
Comment (unaussprechlich) [inline src/main/java/eladkay/hudpixel/modulargui/components/ArrowCounterModularGuiProvider.java:39]: Not all gamemodes have Arrows

File: src/main/java/eladkay/hudpixel/modulargui/components/CCDefusalModularGuiProvider.java [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+        try {
+            if (!doesMatchForGame()) {
+                return;
+            }
+
+            bombStatus = -1;
+
+            EntityPlayer p = getCarrier();
+
+            if (!(p == null)) {
```
Comment (unaussprechlich) [inline src/main/java/eladkay/hudpixel/modulargui/components/CCDefusalModularGuiProvider.java]: p != null :P

File: src/main/java/eladkay/hudpixel/modulargui/components/CCKillstreakModularGuiProvider.java [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+    }
+
+    @Override
+    public void onChatMessage(String textMessage, String formattedMessage) {
+        if(DEATH_PATTERN.matcher(textMessage).find()){
+            onDeath();
+        } else if (KILL_PATTERN.matcher(textMessage).find()){
+            onKill();
+        }
```
Comment (unaussprechlich) [inline src/main/java/eladkay/hudpixel/modulargui/components/CCKillstreakModularGuiProvider.java]: You could have saved so many brackets everywhere :P - it's just my personal styleguide so you don't have to do that

if(DEATH_PATTERN.matcher(textMessage).find())     onDeath();
else if (KILL_PATTERN.matcher(textMessage).find()) onKill();

File: src/main/java/eladkay/hudpixel/modulargui/components/CCTDMModularGuiProvider.java [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+    @Override
+    public void onTickUpdate() {
+        isTDM = ScoreboardReader.getScoreboardTitle().contains("DEATHMATCH");
+    }
+
+    @Override
+    public void onChatMessage(String textMessage, String formattedMessage) {
+        if(textMessage.contains("(Challenge)") || textMessage.contains(FMLClientHandler.instance().getClient().thePlayer.getName() + " has completed a challenge")){
+            challengeKills++;
```
Comment (unaussprechlich) [inline src/main/java/eladkay/hudpixel/modulargui/components/CCTDMModularGuiProvider.java:72]: You can add a static object in the HudPixelMod class, so we don't have to do that every single time :P

File: src/main/java/eladkay/hudpixel/util/HudPixelMethodHandles.java [hunk 6]
```diff
@@ -84,11 +84,9 @@
             Field entityF = ReflectionHelper.findField(Entity.class, "field_70178_ae", "isImmuneToFire", "ab");
             entityImmuneToFire = publicLookup().unreflectSetter(entityF);
             //test
-            Field rightC = ReflectionHelper.findField(Minecraft.class, "rightClickDelayTimer");
-            rightClickDelayTimer = publicLookup().unreflectSetter(rightC);
-
-            Field dirServerResourceP = ReflectionHelper.findField(ResourcePackRepository.class, "dirServerResourcepacks");
-            dirServerResourcepacks  = publicLookup().unreflectSetter(dirServerResourceP);
+            //TODO: FIX THIS, THROWS ERROR
```
Comment (unaussprechlich) [inline src/main/java/eladkay/hudpixel/util/HudPixelMethodHandles.java]: Eld has fixed that, so please update this file!

**Review by unaussprechlich — CHANGES_REQUESTED** [review]
Hey, good work just some small tweaks :P

File: src/main/java/eladkay/hudpixel/modulargui/components/CCDefusalModularGuiProvider.java [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+import eladkay.hudpixel.util.ScoreboardReader;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraftforge.fml.client.FMLClientHandler;
+import net.unaussprechlich.hudpixelextended.util.McColorHelper;
+
+import java.util.regex.Pattern;
+
+public class CCDefusalModularGuiProvider extends SimpleHudPixelModularGuiProvider implements McColorHelper {
+
+    @ConfigPropertyBoolean(category = CCategory.HUD, id = "copsAndCrimsDefusalAddons", comment = "Cops and Crims Defusal Addons", def = true)
```
Comment (Eladkay) [inline src/main/java/eladkay/hudpixel/modulargui/components/CCDefusalModularGuiProvider.java:17]: "Addons" should be be verbose.

File: src/main/java/eladkay/hudpixel/modulargui/components/CCTDMModularGuiProvider.java [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+    }
+
+    @Override
+    public void onGameEnd() {
+        challengeKills = 0;
+    }
+
+    @Override
+    public void onTickUpdate() {
+        isTDM = ScoreboardReader.getScoreboardTitle().contains("DEATHMATCH");
```
Comment (Eladkay) [inline src/main/java/eladkay/hudpixel/modulargui/components/CCTDMModularGuiProvider.java]: Do you really have to do this every tick?

File: src/main/java/eladkay/hudpixel/util/HudPixelMethodHandles.java (hunk 6 above)
Comment (Eladkay) [reply src/main/java/eladkay/hudpixel/util/HudPixelMethodHandles.java]: What error does it throw?

**Review by Eladkay — CHANGES_REQUESTED** [review]

File: src/main/java/eladkay/hudpixel/modulargui/components/CCDefusalModularGuiProvider.java (hunk 7 above)
Comment (FalseHonesty) [reply src/main/java/eladkay/hudpixel/modulargui/components/CCDefusalModularGuiProvider.java:17]: What do you mean?

File: src/main/java/eladkay/hudpixel/modulargui/components/CCTDMModularGuiProvider.java (hunk 5 above)
Comment (FalseHonesty) [reply src/main/java/eladkay/hudpixel/modulargui/components/CCTDMModularGuiProvider.java:72]: Do you mean add the Client object to the HudPixelMod class? If so, you can do it, I'm not familiar enough with Kotlin to do that :stuck_out_tongue:

Comment (unaussprechlich) [reply src/main/java/eladkay/hudpixel/modulargui/components/CCTDMModularGuiProvider.java:72]: I mean the username

**Review by unaussprechlich — APPROVED** [review]

File: src/main/java/eladkay/hudpixel/modulargui/components/CCDefusalModularGuiProvider.java (hunk 7 above)
Comment (Eladkay) [reply src/main/java/eladkay/hudpixel/modulargui/components/CCDefusalModularGuiProvider.java:17]: What do you mean by "addons"? Have a more detailed line there.

File: src/main/java/eladkay/hudpixel/modulargui/components/ArrowCounterModularGuiProvider.java (hunk 1 above)
Comment (Eladkay) [reply src/main/java/eladkay/hudpixel/modulargui/components/ArrowCounterModularGuiProvider.java:15]: Yeah change the id

**Comment by Eladkay** [comment]
Please make it a habit to write with good, consistent code style.
I consistently see code style inconsistencies, especially with bracket spacing.
The correct code style for HudPixel is "if(a) {", note no space after if and space before the last character. Please configure your ide to automatically deal with that.

**Comment by FalseHonesty** [comment]
Will do, thank you.

**Pull request closed without merging**